File: 151-reverse-words-in-a-string/reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        int n=s.length();
        string str="";
        string ans="";
        for(int i=0;i<n;i++)
        {
            if(s[i]==' ' || i==n-1)
            {
                if(i==n-1 && s[i]!=' ')
                {
                    str=str+s[i];
                }
                if(str.length()==0)
                {
                    continue;
                }
                str=str+" ";
                ans=str+ans;
                str="";
            }
            else{
                str=str+s[i];
            }
        }
        ans.pop_back();
        return ans;
    }
};

File: 151-reverse-words-in-a-string/reverse-words-in-a-string.cs
using System;
using System.Collections.Generic;

public class Solution {
    public string ReverseWords(string s) {
        s = s.Trim();
        List<string> words = new List<string>();
        int i = 0;

        while (i < s.Length) {
            while (i < s.Length && s[i] == ' ') i++; 

            if (i >= s.Length) break;

            int start = i;
            while (i < s.Length && s[i] != ' ') i++; 

            words.Add(s.Substring(start, i - start));
        }

        words.Reverse();
        return string.Join(" ", words);
    }
}
